apps/src/templates/projects/projectsModule.js

```diff
@@ -15,7 +21,52 @@ export function selectedGallery(state, action) {
   }
 }
 
+export function projectLists(state, action) {
+  // A map from project type to array of projects, e.g.:
+  //   {
+  //     applab: [{...}, {...}, {...}]
+  //     gamelab: [{...}, {...}, {...}]
+  //     playlab: [{...}, {...}, {...}]
+  //     artist: [{...}, {...}, {...}]
+  //   }
+  state = state || {};
+  switch (action.type) {
+    case SET_PROJECT_LISTS:
+      return action.projectLists;
+    case ADD_OLDER_PROJECTS: {
+      // Append the incoming list of older projects to the existing list,
+      // removing duplicates.
+      const {projects, projectType} = action;
+      state = {...state};
+      state[projectType] = _.unionBy(state[projectType], projects, 'channel');
+      return state;
```

Comment:
That is one of the big benefits. It also provides some nice helpers for being able to some of these more complex merges. i.e. I don't know exactly how `unionBy` works, but that might be easier in an Immutable world.

It might also be the case that this doesn't justify using Immutable.js